protocols/raft/src/main/java/io/atomix/protocols/raft/partition/RaftPartition.java

```diff
@@ -300,6 +294,8 @@ public RaftPartitionServer getServer() {
   }
 
   private void onFailure() {
-    server.stop();
+    final List<CompletableFuture<Void>> listenerFutures =
+        raftFailureListeners.stream().map(l -> l.onRaftFailed()).collect(Collectors.toList());
+    listenerFutures.forEach(future -> future.join());
```

Comment:
```suggestion
    CompletableFuture.allOf(
            raftFailureListeners.stream()
                .map(RaftFailureListener::onRaftFailed)
                .toArray(CompletableFuture[]::new))
        .join();
```

Reply:
This is only applicable if we don't care about the order in which the listeners are completed - which I assumed we didn't since this is a set.